src/Rings/mpoly.jl

```diff
@@ -161,7 +161,12 @@ using .Orderings
 #type for orderings, use this...
 #in general: all algos here needs revision: do they benefit from gb or not?
 
-default_ordering(R::MPolyRing) = degrevlex(R)
+function default_ordering(R::MPolyRing)
+  return get_attribute!(R, :default_ordering) do
+    degrevlex(R)
+  end
+end
```

Comment:
Just a note, this would have been a typical usecase for the `@attr` macro -- or rather `@attr RETURN_TYPE` to make it type stable (this code now is not type stable for this reason)

Reply:
Could you give an example how we should do that alterative?